dask_sql/physical/rel/custom/create_table_as.py

```diff
@@ -32,29 +34,52 @@ class CreateTableAsPlugin(BaseRelPlugin):
     Nothing is returned.
     """
 
-    class_name = "com.dask.sql.parser.SqlCreateTableAs"
+    class_name = ["CreateMemoryTable", "CreateView"]
 
-    def convert(
-        self, sql: "org.apache.calcite.sql.SqlNode", context: "dask_sql.Context"
-    ) -> DataContainer:
-        schema_name, table_name = context.fqn(sql.getTableName())
+    def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContainer:
+        # Rust create_memory_table instance handle
+        create_memory_table = rel.create_memory_table()
+
+        # can we avoid hardcoding the schema name?
+        schema_name, table_name = context.schema_name, create_memory_table.getName()
 
         if table_name in context.schema[schema_name].tables:
-            if sql.getIfNotExists():
+            if create_memory_table.getIfNotExists():
                 return
-            elif not sql.getReplace():
+            elif not create_memory_table.getOrReplace():
                 raise RuntimeError(
                     f"A table with the name {table_name} is already present."
                 )
 
-        sql_select = sql.getSelect()
-        persist = bool(sql.isPersist())
+        input_rel = create_memory_table.getInput()
 
         logger.debug(
-            f"Creating new table with name {table_name} and query {sql_select}"
+            f"Creating new table with name {table_name} and logical plan {input_rel}"
         )
 
-        sql_select_query = context._to_sql_string(sql_select)
-        df = context.sql(sql_select_query)
-
-        context.create_table(table_name, df, persist=persist, schema_name=schema_name)
+        dc = RelConverter.convert(input_rel, context=context)
+        select_names = [field for field in input_rel.getRowType().getFieldList()]
+
+        if select_names:
+            # Use FQ name if not unique and simple name if it is unique. If a join contains the same column
+            # names the output col is prepended with the fully qualified column name
+            field_counts = Counter([field.getName() for field in select_names])
+            select_names = [
+                field.getQualifiedName()
+                if field_counts[field.getName()] > 1
+                else field.getName()
+                for field in select_names
+            ]
+
+            cc = dc.column_container
+            cc = cc.rename(
+                {
+                    df_col: select_name
+                    for df_col, select_name in zip(cc.columns, select_names)
+                }
+            )
+            dc = DataContainer(dc.df, cc)
+
+        df = dc.assign()
```

Comment:
This is lifted from `Context.sql` to compute the table from the `LogicalPlan`, since (to my knowledge) there isn't an obvious way to get the generating SQL for both `CreateMemoryTable` and `CreateView` (though the latter has a [`definition`](https://docs.rs/datafusion/latest/datafusion/logical_plan/struct.CreateView.html#structfield.definition) attribute that could be useful here).

Going to explore moving this shared code to a utility function to cut down on duplication